javalib/src/main/scala/java/lang/Throwables.scala

```diff
@@ -37,17 +38,21 @@ private[lang] object StackTrace {
     cache.getOrElseUpdate(ip, makeStackTraceElement(cursor))
 
   @noinline private[lang] def currentStackTrace(): Array[StackTraceElement] = {
-    val cursor = stackalloc[scala.Byte](2048.toUInt)
-    val context = stackalloc[scala.Byte](2048.toUInt)
-    val offset = stackalloc[scala.Byte](8.toUInt)
-    val ip = stackalloc[CUnsignedLongLong]
     var buffer = mutable.ArrayBuffer.empty[StackTraceElement]
-
-    unwind.get_context(context)
-    unwind.init_local(cursor, context)
-    while (unwind.step(cursor) > 0) {
-      unwind.get_reg(cursor, unwind.UNW_REG_IP, ip)
-      buffer += cachedStackTraceElement(cursor, !ip)
+    if (!LinktimeInfo.asanEnabled) {
+      Zone { implicit z =>
+        val cursor = alloc[scala.Byte](2048.toUSize)
```

Comment:
What is the purpose of changing this from `stackalloc` to `alloc`?

Reply:
On 32-bit platforms, the max stack size can be as little as 8KiB, so using `stackalloc` means that there's a fair chance we would overflow. I'll convert the `offset` and `ip` back to `stackalloc` though.

Reply:
Just was curious.  I defer to your judgment. I think @LeeTibbert was commenting on stack size on some network code but I hadn't considered smaller stacks on 32 bit.

Reply:
Yeah might also make sense to just have a once-allocated threadlocal or something, mostly just a quick patch until we have a better solution.